#!/usr/bin/env python

PACKAGE='point_cloud_subsampler'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#       Name            Type      Lvl   Description                             Default   Min     Max
gen.add("min_z",        double_t, 0,    "Minimum z for obstacles (m)",          0.0)
gen.add("max_z",        double_t, 0,    "Maximum z for obstacles (m)",          2.0)
gen.add("free_z",       double_t, 0,    "Published z for free space (m)",       0.0)
gen.add("obstacle_z",   double_t, 0,    "Published z for obstacles (m)",        1.0)
gen.add("resolution",   double_t, 0,    "Distance between output points (m)",   0.25, 0.0)
gen.add("max_dist",     double_t, 0,    "Maximum Euclidean distance from sensor in the plane (m)",  4.0,  0.0)
gen.add("target_frame", str_t,    1,    "Target TF frame",                      "base_footprint")

exit(gen.generate(PACKAGE, PACKAGE, "PointCloudSubSampler"))
